src/west/app/project.py

```diff
@@ -1373,12 +1374,40 @@ def update_submodules(self, project):
             ref = []
             if (cache_dir):
                 submodule_ref = Path(cache_dir, submodule.path)
-                if any(os.scandir(submodule_ref)):
+                # do not break existing behavior: search for the submodule
```

Comment:
"existing" sounds like this belongs to a commit message. git tracks points in time whereas comments in source code should ideally be "timeless".

There are exceptions but they need context and I miss the context here.